### ISSUE TITLE
Multiple multi-level collection/reference Include can fail

### ISSUE THREAD

**Issue by ajcvickers** [aspnet/EntityFramework#1452]
Repro code:
```
using (var context = CreateContext())
{
    var root = context.Roots
        .Include(e => e.RequiredChildren).ThenInclude(e => e.Children)
        .Include(e => e.OptionalChildren).ThenInclude(e => e.Children)
        .Include(e => e.RequiredSingle).ThenInclude(e => e.Single)
        .Include(e => e.RequiredNonPkSingle).ThenInclude(e => e.Single)
        .Include(e => e.OptionalSingle).ThenInclude(e => e.Single)
        .Single();
}
```

Generated SQL:
```
SELECT TOP(@p0) [e].[AlternateId], [e].[Id], [r].[Id], [r0].[Id], [r1].[Id], [r1].[RootId], [r2].[BackId], [r2].[Id], [o].[Id], [o].[RootId], [o0].[BackId], [o0].[Id]
FROM [Root] AS [e]
LEFT JOIN [RequiredSingle1] AS [r] ON [r].[Id] = [e].[Id]
LEFT JOIN [RequiredSingle2] AS [r0] ON [r0].[Id] = [r].[Id]
LEFT JOIN [RequiredNonPkSingle1] AS [r1] ON [r1].[RootId] = [e].[Id]
LEFT JOIN [RequiredNonPkSingle2] AS [r2] ON [r2].[BackId] = [r1].[Id]
LEFT JOIN [OptionalSingle1] AS [o] ON [o].[RootId] = [e].[Id]
LEFT JOIN [OptionalSingle2] AS [o0] ON [o0].[BackId] = [o].[Id]
ORDER BY [e].[Id]

SELECT [o].[Id], [o].[ParentId]
FROM [Optional1] AS [o]
INNER JOIN (
    SELECT DISTINCT TOP(@p0) [e].[Id]
    FROM [Root] AS [e]
) AS [e] ON [o].[ParentId] = [e].[Id]
ORDER BY [e].[Id], [o].[Id] AS [Id0]
	SqlServerGraphUpdatesTest.cs(35,0)
```

Entities and context, etc.:
```
protected class Root
{
    public int Id { get; set; }
    public Guid AlternateId { get; set; }

    public ICollection<Required1> RequiredChildren { get; set; } = new List<Required1>();
    public ICollection<Optional1> OptionalChildren { get; set; } = new List<Optional1>();
    public RequiredSingle1 RequiredSingle { get; set; }
    public RequiredNonPkSingle1 RequiredNonPkSingle { get; set; }
    public OptionalSingle1 OptionalSingle { get; set; }
}

protected class Required1
{
    public int Id { get; set; }

    public int ParentId { get; set; }
    public Root Parent { get; set; }

    public ICollection<Required2> Children { get; set; } = new List<Required2>();
}

protected class Required2
{
    public int Id { get; set; }

    public int ParentId { get; set; }
    public Required1 Parent { get; set; }
}

protected class Optional1
{
    public int Id { get; set; }

    public int? ParentId { get; set; }
    public Root Parent { get; set; }

    public ICollection<Optional2> Children { get; set; } = new List<Optional2>();
}

protected class Optional2
{
    public int Id { get; set; }

    public int? ParentId { get; set; }
    public Optional1 Parent { get; set; }
}

protected class RequiredSingle1
{
    public int Id { get; set; }

    public Root Root { get; set; }
    public RequiredSingle2 Single { get; set; }
}

protected class RequiredSingle2
{
    public int Id { get; set; }

    public RequiredSingle1 Back { get; set; }
}

protected class RequiredNonPkSingle1
{
    public int Id { get; set; }

    public int RootId { get; set; }
    public Root Root { get; set; }

    public RequiredNonPkSingle2 Single { get; set; }
}

protected class RequiredNonPkSingle2
{
    public int Id { get; set; }

    public int BackId { get; set; }
    public RequiredNonPkSingle1 Back { get; set; }
}

protected class OptionalSingle1
{
    public int Id { get; set; }

    public int? RootId { get; set; }
    public Root Root { get; set; }

    public OptionalSingle2 Single { get; set; }
}

protected class OptionalSingle2
{
    public int Id { get; set; }

    public int? BackId { get; set; }
    public OptionalSingle1 Back { get; set; }
}

protected class GraphUpdatesContext : DbContext
{
    public GraphUpdatesContext(IServiceProvider serviceProvider, DbContextOptions options)
        : base(serviceProvider, options)
    {
    }

    public DbSet<Root> Roots { get; set; }
    public DbSet<RequiredSingle2> RequiredSingle2s { get; set; }
    public DbSet<RequiredNonPkSingle2> RequiredNonPkSingle2s { get; set; }
    public DbSet<OptionalSingle2> OptionalSingle2s { get; set; }
    public DbSet<Required1> Required1s { get; set; }
    public DbSet<Optional1> Optional1s { get; set; }
    public DbSet<Required2> Required2s { get; set; }
    public DbSet<Optional2> Optional2s { get; set; }

    protected virtual void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.Entity<Root>(b =>
            {
                b.HasMany(e => e.RequiredChildren)
                    .WithOne(e => e.Parent)
                    .ForeignKey(e => e.ParentId);

                b.HasMany(e => e.OptionalChildren)
                    .WithOne(e => e.Parent)
                    .ForeignKey(e => e.ParentId);

                b.HasOne(e => e.RequiredSingle)
                    .WithOne(e => e.Root)
                    .ForeignKey<RequiredSingle1>(e => e.Id);

                b.HasOne(e => e.OptionalSingle)
                    .WithOne(e => e.Root)
                    .ForeignKey<OptionalSingle1>(e => e.RootId);

                b.HasOne(e => e.RequiredNonPkSingle)
                    .WithOne(e => e.Root)
                    .ForeignKey<RequiredNonPkSingle1>(e => e.RootId);
            });

        modelBuilder.Entity<Required1>()
            .HasMany(e => e.Children)
            .WithOne(e => e.Parent)
            .ForeignKey(e => e.ParentId);

        modelBuilder.Entity<Optional1>()
            .HasMany(e => e.Children)
            .WithOne(e => e.Parent)
            .ForeignKey(e => e.ParentId);

        modelBuilder.Entity<RequiredSingle1>()
            .HasOne(e => e.Single)
            .WithOne(e => e.Back)
            .ForeignKey<RequiredSingle2>(e => e.Id);

        modelBuilder.Entity<OptionalSingle1>()
            .HasOne(e => e.Single)
            .WithOne(e => e.Back)
            .ForeignKey<OptionalSingle2>(e => e.BackId);

        modelBuilder.Entity<RequiredNonPkSingle1>()
            .HasOne(e => e.Single)
            .WithOne(e => e.Back)
            .ForeignKey<RequiredNonPkSingle2>(e => e.BackId);
    }
}
```
To seed:
```
context.ChangeTracker.AttachGraph(CreateFullGraph());
context.SaveChanges();

protected static Root CreateFullGraph()
{
    return new Root
        {
            RequiredChildren = new List<Required1>
                {
                    new Required1
                        {
                            Children = new List<Required2>
                                {
                                    new Required2(),
                                    new Required2()
                                }
                        },
                    new Required1
                        {
                            Children = new List<Required2>
                                {
                                    new Required2(),
                                    new Required2()
                                }
                        }
                },
            OptionalChildren = new List<Optional1>
                {
                    new Optional1
                        {
                            Children = new List<Optional2>
                                {
                                    new Optional2(),
                                    new Optional2()
                                }
                        },
                    new Optional1
                        {
                            Children = new List<Optional2>
                                {
                                    new Optional2(),
                                    new Optional2()
                                }
                        }
                },
            RequiredSingle = new RequiredSingle1
                {
                    Single = new RequiredSingle2()
                },
            OptionalSingle = new OptionalSingle1
                {
                    Single = new OptionalSingle2()
                },
            RequiredNonPkSingle = new RequiredNonPkSingle1
                {
                    Single = new RequiredNonPkSingle2()
                }
        };
}
```

**Comment by anpete** [comment]
@maumar Looks like a problem in column aliasing...

ORDER BY [e].[Id], [o].[Id] **AS [Id0]**

**Comment by maumar** [comment]
will take a look

**Comment by maumar** [comment]
fixed in 6d918d2472ace2a62fe3dfc83c897f5074a16fe2
